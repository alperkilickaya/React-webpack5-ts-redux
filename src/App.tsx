import React, { Suspense } from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { history } from '@/store';
import { ConnectedRouter } from 'connected-react-router';
import { ToastContainer } from 'react-toastify';
import { Helmet } from 'react-helmet-async';

import Layout from '@/layouts';
import PageLoading from '@/components/page-loading';

// multi language
import '@/locales/i18n';

// load app SCSS styles
import '@/styles/App.scss';

// load Toast styles
import 'react-toastify/dist/ReactToastify.css';

const ReactApp: React.FC = () => {
  return (
    <Router>
      <Helmet titleTemplate="%s - React Starter" defaultTitle="React Starter">
        <meta name="description" content="A React Starter application" />
      </Helmet>

      <ConnectedRouter history={history}>
        <Suspense fallback={<PageLoading show />}>
          <Layout />
          <PageLoading />
        </Suspense>
      </ConnectedRouter>

      <ToastContainer />
    </Router>
  );
};

export default ReactApp;
