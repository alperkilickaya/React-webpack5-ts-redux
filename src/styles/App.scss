@import './size';

@import './container';

@import './font';
@import './mixin';
@import './color';
@import './utils';
@import './reset';
@import './typography.scss';

// load all component style
@import '../components/**/*.scss';

// load all pages style
@import '../features/**/*.scss';

// load all layouts style
@import '../layouts/**/*.scss';
