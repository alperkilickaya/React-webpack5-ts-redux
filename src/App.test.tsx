import React from 'react';
import { createRenderer } from 'react-test-renderer/shallow';
import App from './App';

const renderer = createRenderer();

it('renders <App/> without crashing', () => {
  renderer.render(<App />);
  const renderedOutput = renderer.getRenderOutput();
  expect(renderedOutput).toMatchSnapshot();
});
