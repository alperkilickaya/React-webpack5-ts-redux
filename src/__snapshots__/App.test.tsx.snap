// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <App/> without crashing 1`] = `
<BrowserRouter>
  <Helmet
    defaultTitle="React Starter"
    defer={true}
    encodeSpecialCharacters={true}
    prioritizeSeoTags={false}
    titleTemplate="%s - React Starter"
  >
    <meta
      content="A React Starter application"
      name="description"
    />
  </Helmet>
  <Memo(Connect(ConnectedRouterWithContext))
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <React.Suspense
      fallback={
        <PageLoading
          show={true}
        />
      }
    >
      <Layout />
      <PageLoading />
    </React.Suspense>
  </Memo(Connect(ConnectedRouterWithContext))>
  <ToastContainer
    autoClose={5000}
    closeButton={[Function]}
    closeOnClick={true}
    draggable={true}
    draggableDirection="x"
    draggablePercent={80}
    hideProgressBar={false}
    newestOnTop={false}
    pauseOnFocusLoss={true}
    pauseOnHover={true}
    position="top-right"
    role="alert"
    rtl={false}
    transition={[Function]}
  />
</BrowserRouter>
`;
